src/main/java/filter/lexer/Lexer.java

```diff
@@ -26,7 +26,7 @@
 
             new Rule("\\(", TokenType.LBRACKET),
             new Rule("\\)", TokenType.RBRACKET),
-            new Rule("\\\"", TokenType.QUOTE),
+            new Rule("\\\"[^\"]+\\\"", TokenType.QUOTED_CONTENT),
```

Comment:
The escaping for the surrounding quotes is not needed. This can be simplified to `new Rule("\"[^\"]+\"", ...)`.